src/Ae.js

```diff
@@ -84,30 +84,36 @@ export default class Ae {
     rawTxHex: string,
     networkId: string = "ae_mainnet"
   ): Promise<string> {
+    let isNewAppVersion = true;
+    const appConfig = await this.transport.send(CLA, GET_APP_CONFIGURATION, 0x00, 0x00);
+    if (appConfig[1] === 0 && appConfig[2] < 4) {
+      isNewAppVersion = false;
+    }
     let offset = 0;
+    const indent = 4 + 1 + (isNewAppVersion ? 4 : 0);
     const rawTx = new Buffer(rawTxHex, "hex");
     const networkIdBuffer = new Buffer(networkId);
     const toSend = [];
     while (offset !== rawTx.length) {
       const maxChunkSize =
-        offset === 0 ? 150 - 4 - 4 - 1 - networkIdBuffer.length : 150;
+        offset === 0 ? 150 - indent - networkIdBuffer.length : 150;
       const chunkSize =
         offset + maxChunkSize > rawTx.length
           ? rawTx.length - offset
           : maxChunkSize;
       const buffer = new Buffer(
         offset === 0
-          ? 4 + 4 + 1 + networkIdBuffer.length + chunkSize
+          ? indent + networkIdBuffer.length + chunkSize
           : chunkSize
       );
       if (offset === 0) {
         buffer.writeUInt32BE(accountIndex, 0);
-        buffer.writeUInt32BE(rawTx.length, 4);
-        buffer.writeUInt8(networkIdBuffer.length, 8);
-        networkIdBuffer.copy(buffer, 4 + 4 + 1, offset, networkIdBuffer.length);
+        if (isNewAppVersion) buffer.writeUInt32BE(rawTx.length, 4);
+        buffer.writeUInt8(networkIdBuffer.length, isNewAppVersion ? 8 : 4);
+        networkIdBuffer.copy(buffer, indent, offset, networkIdBuffer.length);
```

Comment:
I think we cam simplify offset calculation by using return values of `write...` functions:
```js
let offset = 0;
offset += buffer.writeUInt32BE(accountIndex, offset);
if (isNewAppVersion) offset += buffer.writeUInt32BE(rawTx.length, offset);
offset += buffer.writeUInt8(networkIdBuffer.length, offset);
offset += networkIdBuffer.copy(buffer, offset, 0, networkIdBuffer.length);
rawTx.copy(buffer, offset, 0, 150 - offset);
```
https://nodejs.org/api/buffer.html#buffer_buf_writeuint32be_value_offset
https://nodejs.org/api/buffer.html#buffer_buf_copy_target_targetstart_sourcestart_sourceend

Reply:
this is not quite correct, but we can do it by using return values thats true.